backend/pkg/connect/create_connector_integration_test.go

```diff
@@ -0,0 +1,378 @@
+		Enabled: true,
+		Clusters: []config.ConnectCluster{
+			{
+				Name: "redpanda_connect",
+				URL:  "http://" + connectHost + ":" + connectPort.Port(),
+			},
+		},
+	}, log)
+
+	require.NoError(err)
+
+	s.connectSvs = connectSvs
+}
+
+func (s *ConnectIntegrationTestSuite) TearDownSuite() {
+	t := s.T()
+	assert := require.New(t)
+
+	assert.NoError(s.redpandaContainer.Terminate(context.Background()))
+	assert.NoError(s.httpBinContainer.Terminate(context.Background()))
+	assert.NoError(s.connectContainer.Terminate(context.Background()))
+	assert.NoError(s.commonNetwork.Remove(context.Background()))
+}
+
+func (s *ConnectIntegrationTestSuite) TestCreateConnector() {
+	t := s.T()
+	require := require.New(t)
+	assert := assert.New(t)
+
+	ctx := context.Background()
+
+	t.Run("HTTP input happy path", func(t *testing.T) {
+		res, connectErr := s.connectSvs.CreateConnector(ctx, "redpanda_connect", connect.CreateConnectorRequest{
+			Name: "http_connect_input",
+			Config: map[string]interface{}{
+				"connector.class":                           "com.github.castorm.kafka.connect.http.HttpSourceConnector",
+				"header.converter":                          "org.apache.kafka.connect.storage.SimpleHeaderConverter",
+				"http.request.url":                          "http://httpbin:80/uuid",
+				"http.timer.catchup.interval.millis":        "10000",
+				"http.timer.interval.millis":                "10000",
+				"kafka.topic":                               "httpbin-input",
+				"key.converter":                             "org.apache.kafka.connect.json.JsonConverter",
+				"key.converter.schemas.enable":              "false",
+				"name":                                      "http_connect_input",
+				"topic.creation.default.partitions":         "1",
+				"topic.creation.default.replication.factor": "1",
+				"topic.creation.enable":                     "true",
+				"value.converter":                           "org.apache.kafka.connect.json.JsonConverter",
+				"value.converter.schemas.enable":            "false",
+			},
+		})
+
+		require.Empty(connectErr)
+		assert.NotNil(res)
+
+		timer := time.NewTimer(32 * time.Second)
+		<-timer.C
+
+		cl, err := kgo.NewClient(
+			kgo.SeedBrokers(s.testSeedBroker),
+			kgo.ConsumeTopics("httpbin-input"),
+			kgo.ConsumeResetOffset(kgo.NewOffset().AtStart()),
+		)
+		require.NoError(err)
+
+		defer cl.Close()
+
+		const waitTimeout = 10 * time.Second
+		ctx, cancel := context.WithTimeout(context.Background(), waitTimeout)
+		defer cancel()
+
+		type uuidValue struct {
+			UUID string `json:"uuid"`
+		}
+
+		type recordValue struct {
+			Value string `json:"value"`
+			Key   string
+		}
+
+		records := make([][]byte, 0)
+
+		for {
+			fetches := cl.PollFetches(ctx)
+			if fetches.IsClientClosed() {
+				break
+			}
+
+			errs := fetches.Errors()
+			if len(errs) == 1 && (errors.Is(errs[0].Err, context.DeadlineExceeded) || errors.Is(errs[0].Err, context.Canceled)) {
+				break
+			}
```

Comment:
Can you explain why we want to break here? It seems we are checking for a timeout , so makes sense to not retry forever, but if we just break here we will continue in the test flow no and in the end we will see the assert len of 3 is not fulfilled

Reply:
We use the context timeout set above to terminate the polling of the kafka records. So canceled context or deadline exceeded is the expected way we break out of the `for` loop in this case. I agree it's weird... open to some other suggestions.

Reply:
Not sure I understand...? we haven't collected all the records yet. I think I am missing something?

Reply:
Maybe we should have a chat, may be quicker to clarify my questions. WDYT? Just hit me up when you are ready for it